fraud/nmt.go

```diff
@@ -0,0 +1,79 @@
+// TEMPORARY
+package fraud
```

Comment:
Remind me why we have to duplicate this code? Can't we just use the implementation in core? Because that `Tree()` method that was added but there was no tagged release yet?